homeassistant/components/unifiprotect/entity.py

```diff
@@ -38,27 +38,47 @@ def _async_device_entities(
     klass: type[ProtectDeviceEntity],
     model_type: ModelType,
     descs: Sequence[ProtectRequiredKeysMixin],
+    unadopted_descs: Sequence[ProtectRequiredKeysMixin],
     ufp_device: ProtectAdoptableDeviceModel | None = None,
 ) -> list[ProtectDeviceEntity]:
-    if len(descs) == 0:
+    if len(descs) + len(unadopted_descs) == 0:
         return []
 
     entities: list[ProtectDeviceEntity] = []
     devices = (
         [ufp_device] if ufp_device is not None else data.get_by_types({model_type})
     )
     for device in devices:
+        assert isinstance(device, (Camera, Light, Sensor, Viewer, Doorlock, Chime))
         if not device.is_adopted_by_us:
+            for description in unadopted_descs:
+                entities.append(
+                    klass(
+                        data,
+                        device=device,
+                        description=description,
+                    )
+                )
+                _LOGGER.debug(
+                    "Adding %s entity %s for %s",
+                    klass.__name__,
+                    description.name,
+                    device.display_name,
+                )
             continue
 
-        assert isinstance(device, (Camera, Light, Sensor, Viewer, Doorlock, Chime))
         for description in descs:
             if description.ufp_perm is not None:
```

Comment:
```suggestion
            if (ufp_perms := description.ufp_perm) is None:
               continue
```

Reply:
That does not work unfortunately. The permission check is optional, if it does not exist, it should not skip adding the entity.

This would be a good place to add a match statement once we are on 3.10+ only.